### ISSUE TITLE
Classifier head architecture

### ISSUE THREAD

**Issue by tsoernes** [explosion/spacy-pytorch-transformers#47]
How does it look? Any way to control the number of hidden neurons, or is there none between the language model and the output?

If you pass 'exclusive_classes=False', does the output neurons use sigmoids instead of softmax?

**Comment by tsoernes** [comment]
I think this is a viable solution to my problem:
```
from spacy_pytorch_transformers.model_registry import register_model, get_pytt_last_hidden

from thinc.t2v import Pooling, mean_pool
from thinc.v2v import Affine
from thinc.api import flatten_add_lengths, chain

from spacy._ml import zero_init, logistic

@register_model("sigmoid_last_hidden")
def sigmoid_last_hidden(nr_class, *, exclusive_classes=True, **cfg):
    width = cfg["token_vector_width"]
    return chain(
        get_pytt_last_hidden,
        flatten_add_lengths,
        Pooling(mean_pool),
        zero_init(Affine(nr_class, width, drop_factor=0.0)),
        logistic,
    )
```
Head arch seems to be defined in `spacy_pytorch_transformers.model_registry`, with a softmax (i.e. exclusive classes) as the default.

**Comment by honnibal** [comment]
@tsoernes Yes, that's basically correct. Unfortunately we don't have support for all of the outputs yet (e.g. for all the attentions or all the hidden layers). But you can control the network architecture by registering a model function, which should return a Thinc model. You can use the `thinc.extra.wrappers.PyTorchWrapper` class to wrap a PyTorch model, if you want to define parts of your model in PyTorch.

**Comment by pbakhtin** [comment]
> I think this is a viable solution to my problem:
> 
> ```
> from spacy_pytorch_transformers.model_registry import register_model, get_pytt_last_hidden
> 
> from thinc.t2v import Pooling, mean_pool
> from thinc.v2v import Affine
> from thinc.api import flatten_add_lengths, chain
> 
> from spacy._ml import zero_init, logistic
> 
> @register_model("sigmoid_last_hidden")
> def sigmoid_last_hidden(nr_class, *, exclusive_classes=True, **cfg):
>     width = cfg["token_vector_width"]
>     return chain(
>         get_pytt_last_hidden,
>         flatten_add_lengths,
>         Pooling(mean_pool),
>         zero_init(Affine(nr_class, width, drop_factor=0.0)),
>         logistic,
>     )
> ```
> 
> Head arch seems to be defined in `spacy_pytorch_transformers.model_registry`, with a softmax (i.e. exclusive classes) as the default.

Thanks for this! It should be added as default architecture into registry!
